File: Plugins/engines/org.eclipse.escriptmonkey.scripting.engine.javascript.rhino.samples/samples/scripts/Find_System_Prints.js
/*
 * Thread: UI
 * Menu: Examples > Workbench >Find System Prints
 * Kudos: Bjorn Freeman-Benson & Ward Cunningham & Arthur Daussy
 * Description: {Look into all java file in your workspace to find System.out.println or System.out.print call. And display task into the task view to remove it}
 * License: EPL 1.0
 */
  
function main() {
	
  loadModule("ResourcesModule");
  loadModule("WorkbenchModule");
  
  
  var files = filesMatching(".*\\.java");
  var match;
   
  for each( file in files ) { 
    file.removeMyTasks(  );
    for each( line in file.getLines() ) {
      if (match = line.string.match(/System\.out\.print(ln)? *\(.*\)/)) {
         line.addMyTask( match[0] );
      }
    }
  }
  getActivePage().showView("org.eclipse.ui.views.TaskList");
}

main()

File: Plugins/engines/org.eclipse.escriptmonkey.lang.python.samples/samples/scripts/Find_System_Prints.py
#
# Thread:UI
# Menu: Examples > Find System Prints
# Kudos: Bjorn Freeman-Benson & Ward Cunningham
# Description: {Look into all java file in your workspace to find System.out.println or System.out.print call. And display task into the task view to remove it}
# License: EPL 1.0
#


#Load module to handle workspace resources
resources=loadModule("ResourcesModule")
#load module to handle workbench
loadModule("WorkbenchModule")

files = resources.filesMatching('.*\\.java')
for file in files:
	file.removeMyTasks()
	for line in file.lines:
		if line.getString().find('System.out.print') != -1:
			line.addMyTask(line.getString().strip())
#This constant come from the WorkbenchModule and is declare as static
getActivePage().showView('org.eclipse.ui.views.TaskList')
